### PR TITLE
feat(dapi-grpc): compile proto to rust using prost

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

We need a way to compile protobuf definitions into Rust code.

## What was done?

Implemented compilation using prost library.

## How Has This Been Tested?

Added examples:

* packages/dapi-grpc/clients/core/v0/rust/core_example.rs
* packages/dapi-grpc/clients/platform/v0/rust/platform_example.rs

## Breaking Changes

Requires protoc to be installed on the build machine:

* for Alpine: install from https://github.com/protocolbuffers/protobuf/releases
* for debian/ubuntu: try `apt-get install protobuf-compiler`

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dapi-grpc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+[package]
+name = "dapi-grpc"
+description = "GRPC client for Dash Platform"
+version = "0.25.0-dev.6"
+authors = [
```
Comment (shumkov) [inline packages/dapi-grpc/Cargo.toml:5]: You probably need to synchronize this list with contributors in package.json

File: packages/dapi-grpc/clients/core/v0/nodejs/core_protoc.js [hunk 2]
```diff
@@ -13,39 +13,39 @@
...
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.GetStatusResponse.Version', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.GetTransactionRequest', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.GetTransactionResponse', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.InstantSendLockMessages', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.RawTransactions', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.TransactionsWithProofsRequest', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.TransactionsWithProofsRequest.FromBlockCase', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.TransactionsWithProofsResponse', null, { proto });
-goog.exportSymbol('proto.org.dash.platform.dapi.v0.TransactionsWithProofsResponse.ResponsesCase', null, { proto });
+var global = Function('return this')();
```
Comment (shumkov) [inline packages/dapi-grpc/clients/core/v0/nodejs/core_protoc.js:16]: I think there was a special patch that should be applied to remove `Function('return this')();`
@markin-io knows better.

File: packages/rs-drive/src/drive/query/mod.rs [hunk 3]
```diff
@@ -102,6 +105,11 @@ impl Drive {
             })?;
         match path.as_str() {
```
Comment (shumkov) [inline packages/rs-drive/src/drive/query/mod.rs:106]: Are these changes somehow related to this PR?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dapi-grpc/clients/core/v0/nodejs/core_protoc.js (hunk 2 above)
Comment (markin-io) [reply packages/dapi-grpc/clients/core/v0/nodejs/core_protoc.js:16]: Yes. `scripts/patch-protobuf-js.sh` must be executed after `scripts/build.sh`. 
It's taken care of in `package.json`

File: packages/dapi-grpc/protos/platform/v0/platform.proto [hunk 4]
```diff
@@ -2,20 +2,25 @@ syntax = "proto3";
...
-  rpc getIdentitiesByPublicKeyHashes (GetIdentitiesByPublicKeyHashesRequest) returns (GetIdentitiesByPublicKeyHashesResponse);
-  rpc waitForStateTransitionResult (WaitForStateTransitionResultRequest) returns (WaitForStateTransitionResultResponse);
-  rpc getConsensusParams (GetConsensusParamsRequest) returns (GetConsensusParamsResponse);
-}
-
-message Proof {
-  bytes merkle_proof = 1;
+  rpc broadcastStateTransition(BroadcastStateTransitionRequest)
+      returns (BroadcastStateTransitionResponse);
+  rpc getIdentity(GetSingleItemRequest) returns (SingleItemResponse);
```
Comment (markin-io) [inline packages/dapi-grpc/protos/platform/v0/platform.proto:8]: I'm afraid that we can not change `.proto` specification at this point it time. 
It will affect many things through the whole **client** -> **server** chain, and this is not something we can deal with efficiently until `platform-test-suite` is stabilized. 

cc @QuantumExplorer

**Review by markin-io — CHANGES_REQUESTED** [review]

File: packages/dapi-grpc/scripts/build.sh [hunk 5]
```diff
@@ -62,13 +59,13 @@ pbjs \
...
-           --ts_out="service=grpc-web:$PLATFORM_WEB_OUT_PATH" \
-           -I="$PLATFORM_PROTO_PATH" \
-           "platform.proto"
+  -v "$PLATFORM_WEB_OUT_PATH:$PLATFORM_WEB_OUT_PATH" \
+  --rm \
+  "$PROTOC_IMAGE" \
+  --js_out="import_style=commonjs:$PLATFORM_WEB_OUT_PATH" \
+  --ts_out="service=grpc-web:$PLATFORM_WEB_OUT_PATH" \
+  -I="$PLATFORM_PROTO_PATH" \
+  "platform.proto"
```
Comment (shumkov) [inline packages/dapi-grpc/scripts/build.sh:68]: @markin-io To make sure that patch won't be missed shouldn't we just move it here instead of having additional call in package.json (`... && yarn exec scripts/patch-protobuf-js.sh`)

Comment (markin-io) [reply packages/dapi-grpc/scripts/build.sh:71]: @lklimek added it to the bottom of the script, so all good

**Review by shumkov — APPROVED** [review]
👍

**Review by markin-io — APPROVED** [review]
